eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -723,6 +777,7 @@ void register() {
     @PreDestroy
     @Override
     public void shutdown() {
+        ApplicationInfoManager.getInstance().unregisterStatusChangeListener(statusChangeListener.getId());
```

Comment:
Yes it is:). But cleaning up the statics (there are many others) should be a separate task so as not to conflate the pull requests.